diesel/src/serialize.rs

```diff
@@ -30,43 +30,34 @@ pub enum IsNull {
 }
 
 /// Wraps a buffer to be written by `ToSql` with additional backend specific
-/// utilities.
-#[derive(Clone, Copy)]
+/// utilities.y
 pub struct Output<'a, T, DB>
 where
     DB: TypeMetadata,
     DB::MetadataLookup: 'a,
 {
     out: T,
-    metadata_lookup: Option<&'a DB::MetadataLookup>,
+    metadata_lookup: Option<&'a mut DB::MetadataLookup>,
 }
 
 impl<'a, T, DB: TypeMetadata> Output<'a, T, DB> {
     /// Construct a new `Output`
-    pub fn new(out: T, metadata_lookup: &'a DB::MetadataLookup) -> Self {
+    pub fn new(out: T, metadata_lookup: &'a mut DB::MetadataLookup) -> Self {
         Output {
             out,
             metadata_lookup: Some(metadata_lookup),
         }
     }
 
-    /// Create a new `Output` with the given buffer
-    pub fn with_buffer<U>(&self, new_out: U) -> Output<'a, U, DB> {
```

Comment:
Yes it's possible to port this method over, but I don't think that would be usable with a mutable connection. Typical usage would be something like [this](https://github.com/diesel-rs/diesel/blob/c0b613024adf970dc888453c984d42e86bad6b3e/diesel/src/pg/types/array.rs#L100-L111). You cannot hold the two `Output` variants there at the same time anymore. You need to deconstruct the inner one as soon as possible. Given that I've opted for removing this method and just "manually" construct the corresponding new temporary `Output` type.

Reply:
It looks like it would be usable, as in example I gave 
> at `diesel/src/pg/types/array.rs`

Because we cannot hold two `Output` functions at the same time I'm also reconstructing an `Output` for each iteration (hence the mutable reference), but just doing it through that helper function.
It did compile with this implementation.

Reply:
Seems like I've missed that. Anyway doing this or just removing this helper function seems to be essentially the same. I would go for the removed function as this reduces the complexity of that type a bit.